File: output/ChatGPT/codenet/Java/C++/s851323333.cpp
#include <iostream>
#include <stack>
using namespace std;

int main() {
    stack<int> list;
    int a;
    
    while(cin >> a) {
        if(a == 0) {
            cout << list.top() << endl;
            list.pop();
        } else {
            list.push(a);
        }
    }
    
    return 0;
}

File: output/StarCoder/codenet/Java/Python/s851323333.py
import sys

class Stack:
    def __init__(self):
        self.items = []

    def isEmpty(self):
        return self.items == []

    def push(self, item):
        self.items.append(item)

    def pop(self):
        return self.items.pop()

    def peek(self):
        return self.items[len(self.items)-1]

    def size(self):
        return len(self.items)

def main():
    stack = Stack()
    while True:
        line = sys.stdin.readline()
        if not line:
            break
        line = line.strip()
        if line == '0':
            if stack.isEmpty():
                print('error')
            else:
                print(stack.pop())
        else:
            stack.push(int(line))

if __name__ == '__main__':
    main()